server/src/sim.rs

```diff
@@ -96,35 +97,38 @@ impl Sim {
     }
 
     fn snapshot_node(&self, node: NodeId) -> save::EntityNode {
-        let mut ids = Vec::new();
-        let mut character_transforms = Vec::new();
-        let mut character_names = Vec::new();
-        let entities = self.graph_entities.get(node);
-
-        for &entity in entities {
-            // TODO: Handle entities other than characters
-            let mut q = self
-                .world
-                .query_one::<(&EntityId, &Position, &Character)>(entity)
-                .unwrap();
-            let Some((id, pos, ch)) = q.get() else {
+        let mut entities = Vec::new();
+        for &entity in self.graph_entities.get(node) {
+            let Ok(entity) = self.world.entity(entity) else {
+                error!("stale graph entity {:?}", entity);
+                continue;
+            };
+            let Some(id) = entity.get::<&EntityId>() else {
                 continue;
             };
-            ids.push(id.to_bits());
-            postcard_helpers::serialize(pos.local.as_ref(), &mut character_transforms).unwrap();
-            postcard_helpers::serialize(&ch.name, &mut character_names).unwrap();
+            let mut components = Vec::new();
+            if let Some(pos) = entity.get::<&Position>() {
+                components.push((
+                    ComponentType::Position as u64,
+                    postcard_helpers::serialize_to_vec(pos.local.as_ref()).unwrap(),
```

Comment:
The double-serialization with Postcard seems a bit strange here, but I don't see a better way.

Reply:
If/when we end up with a bunch of these, a macro might help. There's far worse things in the world than a little boilerplate, though.

Reply:
Er, you probably meant the nesting. This particular invocation is really just a convenient way to coerce dense data to bytes. We could just as well have used `transmute` or pulled in bytemuck or something, but this tool was already handy.